jinxify/src/components/layout/image-overlay-header.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import Image from "next/image";
+
+// Define the props for the component
+type ImageOverlayHeaderProps = {
+	imageSrc: string;
+	title: string;
+	overlayOpacity?: number;
+};
+
+export default function ImageOverlayHeader({
+	imageSrc,
```

Comment:
We can pass a className prop that overrides certain styles instead of passing a overlayOpacity. 

```tsx
import React from "react";
import Image from "next/image";
import { cn } from "@/lib/utils";

// Define the props for the component
type ImageOverlayHeaderProps = {
    imageSrc: string;
    title: string;
    className?: string; // Class for custom styling
};

export default function ImageOverlayHeader({
    imageSrc,
    title,
    className,
}: ImageOverlayHeaderProps) {
    return (
        <div className="relative">
            <Image
                src={imageSrc}
                width={1200}
                height={200}
                alt="Background Image"
                className="absolute top-0 left-0 w-full h-full object-cover rounded-t-lg"
            />

            {/* Overlay with static opacity */}
            <div 
                className={cn(
                    "absolute inset-0 z-10 rounded-t-lg bg-indigo-700 opacity-85", // Static 85% opacity with Tailwind
                    className // Any custom class will be appended
                )}
            />

            <h1 className="relative p-8 text-secondary text-3xl font-bold mb-1 z-20">
                {title}
            </h1>
        </div>
    );
}
```
```tsx
import ImageOverlayHeader from './ImageOverlayHeader'; // Adjust the import path as needed

function ExampleComponent() {
  return (
    <div>
      <ImageOverlayHeader 
        imageSrc="/path/to/your/image.jpg"
        title="Your Title Here"
        className="opacity-50" // Overriding the default opacity to 50%
      />
    </div>
  );
}
```